Index: lib/python/Products/__init__.py
===================================================================
--- lib/python/Products/__init__.py	(revision 72483)
+++ lib/python/Products/__init__.py	(working copy)
@@ -10,4 +10,9 @@
 # FOR A PARTICULAR PURPOSE
 #
 ##############################################################################
-__ac_permissions__=()
+
+try:
+    __import__('pkg_resources').declare_namespace(__name__)
+except ImportError:
+    from pkgutil import extend_path
+    __path__ = extend_path(__path__, __name__)
