Index: configure
===================================================================
--- configure   (revision 85163)
+++ configure   (working copy)
@@ -18,7 +18,7 @@
 # sys.version) below in "best" to "worst" order, not including the
 # target version.  Up to six acceptable python versions are allowed.
 # Do not include the target version number in this list!
-ACCEPTABLE="2.4.3 2.4.2"
+ACCEPTABLE="2.4.3 2.4.2. 2.4.4c1"
 
 # provide the executable names for all the acceptable versions
 # (and the target version) below
